awscrt/io.py

```diff
@@ -374,7 +374,7 @@ def create_client_with_mtls(cert_buffer, key_buffer):
     @staticmethod
     def create_client_with_mtls_pkcs11(*,
                                        pkcs11_lib: 'Pkcs11Lib',
-                                       user_pin: str,
+                                       user_pin: str = None,
```

Comment:
Subtlety: While users **can** pass None, we don't want that to be the default value. Passing `None` to "log into the token with a protected authentication path" is an uncommon use-case that people should be explicit about

```suggestion
                                       user_pin: Union[str, None],
```

I feel like `Union[str, None]`

you'll also need `from typing import Union` at the top of the file

Newer versions of python would have simply let us write this as `str|None` but in Python 3.7 we need to write it as `Union[str, None]`